internal/services/openai.go

```diff
@@ -0,0 +1,117 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/DIMO-Network/devices-api/internal/config"
+	"github.com/pkg/errors"
+)
+
+type OpenAI interface {
+	QueryDeviceErrorCodes(make, model string, year int32, errorCodes []string) (string, error)
+}
+
+type openAi struct {
+	baseURL    string
+	token      string
+	httpClient *http.Client
+}
+
+type ChatGptResponseChoices struct {
+	Message struct {
+		Role    string
+		Content string
+	}
+	Index        int
+	FinishReason string `json:"finish_reason"`
+}
+
+type ChatGptResponse struct {
+	ID      string
+	Object  string
+	Created int
+	Model   string
+	Usage   struct {
+		PromptTokens     int
+		CompletionTokens int
+		TotalTokens      int
+	}
+	Choices []ChatGptResponseChoices `json:"choices"`
+}
+
+func NewOpenAI(c config.Settings) OpenAI {
+	return openAi{
+		baseURL: c.OpenAiBaseURL,
+		token:   c.OpenAiSecretKey,
+		httpClient: &http.Client{
+			Timeout: 10 * time.Second,
+		},
+	}
+}
+
+func (o openAi) askChatGpt(body *strings.Reader) (*ChatGptResponse, error) {
```

Comment:
```suggestion
func (o openAi) askChatGPT(body io.Reader) (*ChatGptResponse, error) {
```
I think this is all you need. I think the division of labor here is a bit strange, though—I'm surprised the serialization happens outside.